components/merkledb/src/indexes/iter.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Generic iterator types used by all indexes.
+
+use crate::{
+    views::{Iter, RawAccess, View},
+    BinaryKey, BinaryValue,
+};
+
+/// Iterator over key-value pairs of an index.
+///
+/// This structure is returned by the [`IndexIterator`] trait and by inherent methods
+/// of some indexes.
+///
+/// [`IndexIterator`]: trait.IndexIterator.html
+#[derive(Debug)]
+pub struct Entries<'a, K: ?Sized, V> {
```

Comment:
The plural is a little bit confusing here 🤔

Reply:
But this is how iterators are usually named in the standard Rust library. Although, in this particular case, `Iter` is another option.